community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/index/LuceneIndexAccessor.java

```diff
@@ -202,18 +201,25 @@ private void closeIndexResources() throws IOException
     @Override
     public IndexReader newReader()
     {
-        final IndexSearcher searcher = searcherManager.acquire();
-        final TaskControl token = taskCoordinator.newInstance();
-        final Closeable closeable = new Closeable()
+        try
         {
-            @Override
-            public void close() throws IOException
+            final IndexSearcher searcher = searcherManager.acquire();
+            final TaskControl token = taskCoordinator.newInstance();
+            final Closeable closeable = new Closeable()
             {
-                searcherManager.release( searcher );
-                token.close();
-            }
-        };
-        return makeNewReader( searcher, closeable, token );
+                @Override
+                public void close() throws IOException
+                {
+                    searcherManager.release( searcher );
+                    token.close();
+                }
+            };
+            return makeNewReader( searcher, closeable, token );
+        }
+        catch ( IOException e )
+        {
+            throw new RuntimeException( e );
```

Comment:
Is really RuntimeException an appropriate exception here. We should be explicit with exception handling in as many places as possible.